streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamGlobalKTableJoinTest.java

```diff
@@ -246,4 +262,30 @@ public void shouldJoinOnNullKeyWithNonNullKeyMapperValues() {
         processor.checkAndClearProcessResult(new KeyValueTimestamp<>(null, "X0,FKey0+Y0", 0),
                 new KeyValueTimestamp<>(1, "X1,FKey1+Y1", 1));
     }
+
+    @Test
+    public void shouldPerformTimestampedGet() {
```

Comment:
It doesn't look like there's existing integration test coverage for KStream-GlobalKTable joins, so I've added these tests here instead. 

For KStream-KTable joins, there's also `KStreamKTableJoinTest` and `KStreamKTableLeftJoinTest` which I did not update, since I opted to update `StreamTableJoinIntegrationTest` instead. AFAICT both sets of tests are run with the TopologyTestDriver so updating `KStreamKTableJoinTest` and `KStreamKTableLeftJoinTest` would be redundant to having updated `StreamTableJoinIntegrationTest`, but if that's not true then I can update those unit tests as well.